javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js

```diff
@@ -1,272 +1,259 @@
-        this.validScoreOptions["full-house"] = 25;
-      }
-    }
+  if (smallStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(3, 30);
+  } else if (largeStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(4, 40);
+  } else {
+    updateRadioOption(5, 0);
+  }
+};
 
-    if (totalConsecutives >= 4) {
-      this.validScoreOptions["small-straight"] = 30;
-    }
+/**
+ * ! This will help us 3 of a kind / 4 of a kind
+ */
+
+const getHighestDuplicates = (arr) => {
+  const counts = {};
 
-    if (totalConsecutives === 5) {
-      this.validScoreOptions["large-straight"] = 40;
+  // Count the occurrences of each number in the array
+  for (const num of arr) {
+    if (counts[num]) {
+      counts[num]++;
+    } else {
+      counts[num] = 1;
     }
   }
 
-  enableValidScoreInputs() {
-    const availableScoreInputs = scoreInputs;
-    const availableRadioScores = scoreSpans;
-    // I believe this is the first time Object.keys has been introduced
-    const validScoreOptionsKeys = Object.keys(this.validScoreOptions);
-
-    for (let i = 0; i < validScoreOptionsKeys.length; i++) {
-      for (let j = 0; j < availableScoreInputs.length; j++) {
-        if (validScoreOptionsKeys[i] === availableScoreInputs[j].value) {
-          availableScoreInputs[j].disabled = false;
-          availableRadioScores[j].textContent = `, score= ${
-            this.validScoreOptions[availableScoreInputs[j].value]
-          }`;
-        }
-      }
+  // Find the highest count of duplicates that is at least 3 or 4
+  let highestCount = 0;
+  for (const num of arr) {
+    const count = counts[num];
+    if (count >= 3 && count > highestCount) {
+      highestCount = count;
+    }
+    if (count >= 4 && count > highestCount) {
+      highestCount = count;
     }
   }
 
-  isKeepScoreSuccess() {
-    const allScoreInputs = scoreInputs;
+  // Filter the original array to include only the numbers that appear the highest number of times
+  const highestDuplicates = arr.filter((num) => counts[num] === highestCount);
 
-    for (let i = 0; i < allScoreInputs.length; i++) {
-      if (allScoreInputs[i].checked) {
-        const currentValue = allScoreInputs[i].value;
-        const currentScore = this.validScoreOptions[currentValue];
-        this.totalScore += currentScore;
-        this.scoreHistory.push({ [allScoreInputs[i].value]: currentScore });
-        this.currentRound++;
-        this.rollsInCurrentRound = 0;
+  // according to rules, it should be sum of all 5 dices
+  const sumOfAllDices = displayArray.reduce((a, b) => a + b, 0);
 
-        return true;
-      }
-    }
-    alert("Please select a score.");
-    return false;
+  // Return the highest duplicates only if there are at least three or four of them
+  if (highestCount >= 4) {
+    updateRadioOption(1, sumOfAllDices);
+  } else if (highestCount >= 3) {
+    updateRadioOption(0, sumOfAllDices);
+  } else {
+    updateRadioOption(5, 0);
   }
```

Comment:
```suggestion
  if (highestCount >= 4) {
    updateRadioOption(1, sumOfAllDices);
  }
  if (highestCount >= 3) {
    updateRadioOption(0, sumOfAllDices);
  }
  updateRadioOption(5, 0);
```

These should not be mutually exclusive, IMHO.